completions/rsync

```diff
@@ -1,5 +1,35 @@
 # bash completion for rsync                                -*- shell-script -*-
 
+_comp_cmd_rsync__vercomp()
+{
+    if [[ "$1" == "$2" ]]; then
+        return 0
+    fi
+    local IFS=.
+    local i ver1=($1) ver2=($2)
+    # fill empty fields in ver1 with zeros
+    for ((i = ${#ver1[@]}; i < ${#ver2[@]}; i++)); do
+        ver1[i]=0
+    done
+    for ((i = 0; i < ${#ver1[@]}; i++)); do
+        if [[ ! ${ver1[i]} ]]; then
+            # replace empty strings ver1 with zeros
+            ver1[i]=0
+        fi
+        if [[ ! ${ver2[i]} ]]; then
+            # fill empty fields in ver2 with zeros
+            ver2[i]=0
+        fi
```

Comment:
Can we maybe use `${ver1[i]:-0}` when we compare the values (such as `10#${ver1[i]:-0}`)? I'd think it is simpler. Another way is to use `10#0${ver1[i]-}`.